From 27664605b945b13e1d2b71adea822ace2099cc96 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Michal=20=C4=8Ciha=C5=99?= <michal@cihar.com>
Date: Fri, 17 Jun 2016 14:26:05 +0200
Subject: [PATCH] Improve error handling in OpenID example
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

- properly check parameter types
- catch all exceptions (eg. network error)

Signed-off-by: Michal Čihař <michal@cihar.com>
---
 examples/openid.php | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/examples/openid.php b/examples/openid.php
index da4a13fb531..ab7707f5382 100644
--- a/examples/openid.php
+++ b/examples/openid.php
@@ -98,9 +98,9 @@ function Show_page($contents)
 }
 
 /* Grab identifier */
-if (isset($_POST['identifier'])) {
+if (isset($_POST['identifier']) && is_string($_POST['identifier'])) {
     $identifier = $_POST['identifier'];
-} else if (isset($_SESSION['identifier'])) {
+} else if (isset($_SESSION['identifier']) && is_string($_SESSION['identifier'])) {
     $identifier = $_SESSION['identifier'];
 } else {
     $identifier = null;
@@ -109,7 +109,7 @@ function Show_page($contents)
 /* Create OpenID object */
 try {
     $o = new OpenID_RelyingParty($returnTo, $realm, $identifier);
-} catch (OpenID_Exception $e) {
+} catch (Exception $e) {
     $contents = "<div class='relyingparty_results'>\n";
     $contents .= "<pre>" . $e->getMessage() . "</pre>\n";
     $contents .= "</div class='relyingparty_results'>";
